docs/developer-docs/smart-contracts/candid/candid-howto.mdx

```diff
@@ -18,11 +18,50 @@ Now, let’s explore how to interact with this canister in different scenarios w
 
-If you write a canister in Motoko, for example, the compiler automatically generates a Candid description when you compile the program. If you use the SDK, you will typically see the auto-generated `.did` files in the `/declarations` directory of your project. Since these files are auto-generated, it is recommended that they should not be manually edited. Even if you change the `.did` files in your project, they will be overwritten in the next `dfx build`.
+### Auto-generated `.did` files
+
+If you write a canister in Motoko, for example, the compiler automatically generates a Candid description when you compile the program. If you use `dfx`, you will typically see the auto-generated `.did` files in the `/declarations` directory of your project. Since these files are auto-generated, it is recommended that they should not be manually edited. Even if you change the `.did` files in your project, they will be overwritten in the next `dfx build`.
 
 For canisters written in Rust, the Candid extractor tool can be used with the IC CDK version `0.11.0` and newer. You can find the full instructions for using the Candid extractor [here](/docs/current/developer-docs/backend/candid/generating-candid).
 
-In other languages, you will have to write the Candid interface description manually. With the help of types, tools have been developed to automatically generate UI and perform random testing based on the service description file.
+### Writing `.did` files
+
+In other languages, you will have to write the Candid interface description manually. The most simple interface description can be found below, which defines a service with no public methods:
+
+```
+service : {}
+```
+
+To add a public method, add the method's name, followed by the data types that the method accepts and returns:
+
+```
+service : {
+  ping : () -> ();
+}
+```
+
+In this `ping` method, it does not accept any types and does not return any types.
+
+Here is an example that defines the public method `greet`, which accepts type `text` and returns type `text` using a query call.
+
+```
+service : {
+    "greet" : (text) -> (text) query;
+};
+```
+
+[Learn more about supported data types in the Candid reference documentation](/docs/current/references/candid-ref#supported-types).
+
+### Splitting interface descriptions across multiple files
+
+In some workflows, it may be beneficial to split interface descriptions across multiple `.did` files. For example, one file may be used to define type definitions that are used by several canisters, while another file is used to define the main service for one of those canisters. 
+
+You can import both interface descriptions into your canister with two import statements:
+
+```
+import A "a.did";  // Imports only type definitions
+import service B "b.did";  // Imports both type definitions and main service
+```
```

Comment:
Isn't this Motoko specific? I don't think you can do it in Rust for example.

Reply:
Added a note indicating this is Motoko-specific.